core.go

```diff
@@ -171,6 +171,24 @@ func buildURL(baseURL string, info *Request) string {
 	return strings.Join(tokens, "/")
 }
 
+func buildURLV4(baseURL string, info *Request) string {
+	tokens := []string{baseURL, info.Resource}
```

Comment:
Doesn't look DRY to me.

Could you refactor old `buildURL` and new `buildURLV4` into:
```
func buildURL(baseURL string, info *Request) string {
    buildURLPath([]string{baseURL, apiPath, info})
}
```
```
func buildURLV4(baseURL string, info *Request) string {
    buildURLPath([]string{baseURL, info})
}
```
or something like that?